src/ObjectUtils/ElfFile.cpp

```diff
@@ -425,6 +448,51 @@ ErrorMessageOr<ModuleSymbols> ElfFileImpl<ElfT>::LoadSymbolsFromDynsym() {
   return module_symbols;
 }
 
+template <typename ElfT>
+absl::flat_hash_set<uint64_t> ElfFileImpl<ElfT>::LoadHotpatchableAddresses() {
+  if (!has_patchable_function_entries_section_) {
+    return absl::flat_hash_set<uint64_t>();
+  }
+  const llvm::object::ELFFile<ElfT>& elf_file = object_file_->getELFFile();
+  llvm::Expected<typename ElfT::ShdrRange> sections_or_error = elf_file.sections();
+  if (!sections_or_error) {
+    auto error_message = absl::StrFormat("Unable to load sections: %s",
+                                         llvm::toString(sections_or_error.takeError()));
+    ORBIT_ERROR("%s", error_message);
+    return absl::flat_hash_set<uint64_t>();
+  }
+
+  absl::flat_hash_set<uint64_t> patchable_symbols;
+  for (const typename ElfT::Shdr& section : sections_or_error.get()) {
+    llvm::Expected<llvm::StringRef> name_or_error = elf_file.getSectionName(section);
+    if (!name_or_error) {
+      ORBIT_ERROR("%s", absl::StrFormat("Unable to get section name: %s",
+                                        llvm::toString(name_or_error.takeError())));
+      return absl::flat_hash_set<uint64_t>();
+    }
+    llvm::StringRef name = name_or_error.get();
+    if (name.str() != "__patchable_function_entries") {
+      continue;
+    }
+
+    llvm::Expected<llvm::ArrayRef<uint8_t>> contents_or_error =
+        elf_file.getSectionContents(section);
+    if (!contents_or_error) {
+      ORBIT_ERROR("Could not read __patchable_function_entries section: %s",
+                  llvm::toString(contents_or_error.takeError()));
+      continue;
+    }
+    const llvm::ArrayRef<uint8_t>& contents = contents_or_error.get();
+    const size_t length = contents.size() / sizeof(uint64_t);
+    for (size_t i = 0; i < length; i++) {
```

Comment:
I think this at least deserves a brief comment. Or maybe use some other mechanism than the memcpy, that is a bit more explicit (e.g. have an array<uint64_t, length> and copy the complete array at once. Not sure though).

Reply:
Added a comment and replaced the implementation with something mildly nicer. ptal.

Reply:
Thanks.